File: Tests/FMDBTempDBTests.m
//
//  FMDBTempDBTests.m
//  fmdb
//
//  Created by Graham Dennis on 24/11/2013.
//
//

#import "FMDBTempDBTests.h"

static NSString *const testDatabasePath = @"/tmp/tmp.db";
static NSString *const populatedDatabasePath = @"/tmp/tmp-populated.db";

@implementation FMDBTempDBTests

+ (void)setUp
{
    [super setUp];
    
    // Delete old populated database
    NSFileManager *fileManager = [NSFileManager defaultManager];
    [fileManager removeItemAtPath:populatedDatabasePath error:NULL];
    
    if ([self respondsToSelector:@selector(populateDatabase:)]) {
        FMDatabase *db = [FMDatabase databaseWithPath:populatedDatabasePath];
        
        [db open];
        [self populateDatabase:db];
        [db close];
    }
}

- (void)setUp
{
    [super setUp];
    
    // Delete the old database
    NSFileManager *fileManager = [NSFileManager defaultManager];
    [fileManager removeItemAtPath:testDatabasePath error:NULL];
    
    if ([[self class] respondsToSelector:@selector(populateDatabase:)]) {
        [fileManager copyItemAtPath:populatedDatabasePath toPath:testDatabasePath error:NULL];
    }
    
    self.db = [FMDatabase databaseWithPath:testDatabasePath];
    
    XCTAssertTrue([self.db open], @"Wasn't able to open database");
    [self.db setShouldCacheStatements:YES];
}

- (void)tearDown
{
    [super tearDown];
    
    [self.db close];
}

- (NSString *)databasePath
{
    return testDatabasePath;
}

@end

File: FMDBSwift/Tests/FMDBTests/FMDBTempDBTests.swift
import XCTest
@testable import FMDB

class FMDBTempDBTests: XCTestCase {
//    
//    var db : FMDatabase
//    var databasePath : String?
//    
//    
//    let testDatabasePath = "/tmp/tmp.db"
//    static let populatedDatabasePath = "/tmp/tmp-populated.db"
//    
//    
//    public override init(selector: Selector) {
//        self.db = FMDatabase()
//        super.init(selector: selector)
//    }
//    
//    override static func setUp() {
//        
//        
//        super.setUp()
//        
//        let url = NSURL(fileURLWithPath: populatedDatabasePath, isDirectory: false)
//        
//        let fileManager = FileManager()
//        try? fileManager.removeItem(at: url as URL)
//        
//        
//        /*
//        // Delete old populated database
//        NSFileManager *fileManager = [NSFileManager defaultManager];
//        [fileManager removeItemAtPath:populatedDatabasePath error:NULL];
//        
//        if ([self respondsToSelector:@selector(populateDatabase:)]) {
//            FMDatabase *db = [FMDatabase databaseWithPath:populatedDatabasePath];
//            
//            [db open];
//            [self populateDatabase:db];
//            [db close];
//        }*/
//        
//        
//    }
    
    public func emptyDatabase(path: String) -> FMDatabase {
        let url = URL(fileURLWithPath: path, isDirectory: false)

        let fileManager = FileManager()
        try? fileManager.removeItem(at: url as URL)
        
        let db = FMDatabase.database(with: url)
        
        return db
        
    }

    public func populatedOpenDatabase(path: String) -> FMDatabase {
        let url = URL(fileURLWithPath: path, isDirectory: false)

        let fileManager = FileManager()
        try? fileManager.removeItem(at: url as URL)
        
        let db = FMDatabase.database(with: url)
        
        
        do {
            
            try db.open()
            
            try db.executeUpdate("create table test (a text, b text, c integer, d double, e double)")
            
            try db.beginTransaction()
            
            var i = 0
            while (i < 20) {
                i = i + 1
                
                try db.executeUpdate("insert into test (a, b, c, d, e) values (?, ?, ?, ?, ?)", "hi'", "number \(i)", i, NSDate(), 2.2)
                
                try db.executeUpdate("insert into test (a, b, c, d, e) values (?, ?, ?, ?, ?)", "hi again'", "number \(i)", i, NSDate(), 2.2)
                
            }
            try db.commit()
            
            
            try db.executeUpdate("create table t3 (a somevalue)")
            
            try db.beginTransaction()
            
            i = 0
            while (i < 20) {
                i = i + 1
                
                try db.executeUpdate("insert into t3 (a) values (?)", i)
            }
            try db.commit()
        }
        catch {
            assert(false, "\(error)")
        }
        
        
        
        
        
        
        
        
        
        
        return db
        
    }

    
    
    
    
    
    
    
}
